runtime/filetype.lua

```diff
@@ -10,7 +10,15 @@ vim.api.nvim_create_autocmd({ 'BufRead', 'BufNewFile', 'StdinReadPost' }, {
   group = 'filetypedetect',
   callback = function(args)
     local ft, on_detect = vim.filetype.match({ filename = args.match, buf = args.buf })
-    if not ft then
+    local has_ft = false
+
+    if type(ft) == "nil" then
+      has_ft = not ft
+    elseif type(ft) == "string" then
+      has_ft = #ft == 0
+    end
+
+    if has_ft then
```

Comment:
```suggestion
   if not ft or ft == '' then
```

Reply:
yep .can do this. but as i said the root of issue  we should explicitly give ft a type ? if you accept this check . i will update